services/tag.go

```diff
@@ -365,7 +395,8 @@ func (t *Tag) HashReferenceByTag(
 func (t *Tag) NewTag(ctx context.Context, namespace, name, from string, mirror bool) error {
 	it := &imagtagv1beta1.Tag{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: name,
+			Name:       name,
+			Finalizers: []string{finalizerstr},
```

Comment:
The "finalizer" should, ideally, be added right after the reconciliation happens. For instance, when the tag is successfully reconciled the first time, then it should append a finalizer entry.